### PR TITLE
Test cases for Token api 

### PR DESCRIPTION
**Pull request by nandkishorr**
## Merge Checklist

- [ ] Tests added/fixed
- [ ] Update docs in `/docs`
- [ ] Linting Complete
- [ ] Any other necessary step

_*Only PR's with test cases included and passing lint and test pipelines will be reviewed*_

@ohcnetwork/care-backend-maintainers @ohcnetwork/care-backend-admins

## Summary by CodeRabbit

* **Bug Fixes**
  * Enforced resource/facility consistency when assigning or changing a token's sub-queue and preserved current-token behavior.
  * Tightened validation and adjusted permission checks and denial messages for create, update, retrieve, list and delete actions.
  * set-next now returns proper API responses and authorizes against the specific token instance.

* **Tests**
  * Added comprehensive tests covering token lifecycle, permissions, filtering, and queue/sub-queue/resource edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/ohcnetwork/care/pull/3336?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Review by sainak — COMMENTED** [review]
missing test cases for token update and set_next

File: care/emr/tests/test_token_api.py [hunk 1]
```diff
@@ -0,0 +1,743 @@
...
+            data=token_data,
+            format="json",
+        )
+        self.assertEqual(response.status_code, 400)
+        self.assertIn(
+            "Sub Queue and Queue are not in the same facility",
+            response.data["errors"][0]["msg"],
+        )
+
+    # Testcases for Token update
```
Comment (nandkishorr) [inline care/emr/tests/test_token_api.py:223]: @sainak

**Review by sainak — APPROVED** [review]

**Review by praffq — APPROVED** [review]

**Comment by praffq** [comment]
@nandkishorr  please resolve conflicts

**Review by praffq — APPROVED** [review]

File: care/emr/api/viewsets/scheduling/token.py
Comment (praffq) [inline care/emr/api/viewsets/scheduling/token.py:110]: This is redundent, already checked at 101-104

**Review by praffq — CHANGES_REQUESTED** [review]

**Review by praffq — APPROVED** [review]
